share/completions/pacaur.fish

```diff
@@ -0,0 +1,168 @@
+# Completions for pacaur
+
+set -l progname pacaur
+complete -e -c $progname
+complete -c $progname -f
+
+set -l listinstalled "(pacman -Q | string replace ' ' \t)"
+# This might be an issue if another package manager is also installed (e.g. for containers)
+set -l listall "(__fish_print_packages)"
+set -l listrepos "(__fish_print_pacman_repos)"
+set -l listgroups "(pacman -Sg)\t'Package Group'"
+
+set -l hasopt '__fish_contains_opt -s B backup -s C -s G getpkgbuild -s P pkgbuild -s D database -s Q query -s R remove -s S sync -s U upgrade -s F files stats'
+set -l noopt "not $hasopt"
+
+set -l backup '__fish_contains_opt -s B backup'
+set -l clean '__fish_contains_opt -s C'
+set -l getpkgbuild '__fish_contains_opt -s G getpkgbuild'
+set -l pkgbuild '__fish_contains_opt -s P pkgbuild'
+set -l database '__fish_contains_opt -s D database'
+set -l query '__fish_contains_opt -s Q query'
+set -l remove '__fish_contains_opt -s R remove'
+set -l sync '__fish_contains_opt -s S sync'
+set -l upgrade '__fish_contains_opt -s U upgrade'
+set -l files '__fish_contains_opt -s F files'
+
+# HACK: We only need these three to coerce fish to stop file completion and complete options
+complete -c $progname -n $noopt -a "-D" -d "Modify the package database"
+complete -c $progname -n $noopt -a "-Q" -d "Query the package database"
+complete -c $progname -n $noopt -a "-C" -d "Manage .pac* files"
+# Primary operations
+complete -c $progname -s D -f -l database -n $noopt -d 'Modify the package database'
+complete -c $progname -s Q -f -l query -n $noopt -d 'Query the package database'
+complete -c $progname -s R -f -l remove -n $noopt -d 'Remove packages from the system'
+complete -c $progname -s S -f -l sync -n $noopt -d 'Synchronize packages'
+complete -c $progname -s T -f -l deptest -n $noopt -d 'Check dependencies'
+complete -c $progname -s U -f -l upgrade -n $noopt -d 'Upgrade or add a local package'
+complete -c $progname -s F -f -l files -n $noopt -d 'Query the files database'
+complete -c $progname -s V -f -l version -d 'Display version and exit'
+complete -c $progname -s h -f -l help -d 'Display help'
+# pacaur operations
+complete -c $progname -n $noopt -s s    -l search      -d "(AUR) Search for packages"
+complete -c $progname -n $noopt -s i    -l info        -d "(AUR) Show info for packages"
+complete -c $progname -n $noopt -s d    -l download    -d "(AUR) Clone the packages' build files"
+# @TODO:
+#complete -c $progname -n __fish_contains_opt -s d download -s d -l download -d 'Download recursively'
```

Comment:
From the pacman completion: 

```fish
complete -c $progname -n "$sync; and not __fish_contains_opt -s u sysupgrade" -s u -l sysupgrade -d 'Upgrade all packages that are out of date'
complete -c $progname -n "$sync; and __fish_contains_opt -s u sysupgrade" -s u -l sysupgrade -d 'Also downgrade packages'
```

Unfortunately that seems to not generate `-Suu` properly, only `-Su -u` and the long variant work.